anonymizer.go

```diff
@@ -0,0 +1,40 @@
+package goanonymizer
+
+import (
+	"errors"
+	"net"
+)
+
+//Anonymize struct that hold mask for V4 and V6.
+type Anonymize struct {
+	v4Mask net.IPMask
+	v6Mask net.IPMask
+}
+
+//NewAnonymize returns instance of Anonymize.
+//If v4Mask and v6Mask are not provided, default values are used.
+func NewAnonymize(v4Mask, v6Mask string) *Anonymize {
+	ipV4 := net.ParseIP(v4Mask)
+	if ipV4 == nil {
+		ipV4 = net.ParseIP("255.255.255.0")
+	}
+	ipV6 := net.ParseIP(v6Mask)
+	if ipV6 == nil {
+		ipV6 = net.ParseIP("ffff:ffff:ffff:ffff:0000:0000:0000:0000")
+	}
+
+	return &Anonymize{v4Mask: net.IPMask(ipV4.To4()), v6Mask: net.IPMask(ipV6)}
+}
+
+//AnonymizeIp gets IP and transform it based on v4Mask and v6Mask.
+func (a Anonymize) AnonymizeIp(ipStr string) (string, error) {
```

Comment:
We are forcing the consumers of the library to use strings. I think we should split this function in two:
```go
func (a *IPAnonymizer) Anonymize(ip net.IP) (net.IP, error) {
    ...
}

func (a *IPAnonymizer) AnonymizeString(ipString string) (string, error) {
    // call a.Anonimize
}
```